lib/src/main/java/org/asamk/signal/manager/util/DataURI.java

```diff
@@ -0,0 +1,64 @@
+package org.asamk.signal.manager.util;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Data URI record that follows RFC 2397.
+ *
+ * @param mediaType the media type. If empty, the default media type "text/plain" is used.
+ * @param parameter the list of parameters. Must be URL escaped encoded.
+ * @param data      the data. If base64 is not given, the data is treated as ASCII string.
+ */
+@SuppressWarnings({"java:S6218"})
+public record DataURI(String mediaType, Map<String, String> parameter, byte[] data) {
+
+    public static final Pattern DATA_URI_PATTERN = Pattern.compile(
+            "data:(?<type>.+?\\/.+?)?(?<parameters>;.+?)?(?<base64>;base64)?,(?<data>.+)",
```

Comment:
The Pattern isn't anchored and would also match parts of the input
It should probably use `\A` at the beginning and `\z` at the end.
https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html